### PR TITLE
Add docker nodesets for acceptance tests

### PR DESCRIPTION
**Pull request by Yuav**

### CODE REVIEW

**Comment by dhoppe** [comment]
I think we should concentrate on releases we are already using for Vagrant. For example CentOS 5 and Debian 6 are not used anymore and regarding Ubuntu we only support LTS releases. 

I also think we should use an .erb template for the nodesets, because we might need to get different packages installed.

**Comment by bastelfreak** [comment]
Probably we should also add a comment into each file that it is managed by modulesync.

**Comment by Yuav** [comment]
I'm fine with deprecating old distros, but I disagree that it should be templates. In my opinion the nodesets should be standardized, and any additional packages required can be installed using the acceptance test itself as Puppet code. (E.G adding package { 'git': ensure => present } in the test itself)

Doesn't seem to be comments in the other nodesets (or other files really) with regards to modulesync? It's not sufficient with the commit message from modulesync with gets displayed by GitHub?

**Comment by dhoppe** [comment]
This could be a good solution. @bastelfreak What do you thing?

My bad. CentOS is not EOL until 2017-03-31. I just added some comments at the other nodesets to point to modulesync.

**Comment by dhoppe** [comment]
Could you please restore centos-511-x64.yml, add one for docker and copy the comments from the other nodesets? 

I spoke to @bastelfreak and will merge your pr afterwards.

**Comment by Yuav** [comment]
Fixed

File: moduleroot/spec/acceptance/nodesets/docker/centos-5.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+HOSTS:
```
Comment (dhoppe) [inline moduleroot/spec/acceptance/nodesets/docker/centos-5.yml]: Please prepend:
```
---
# This file is managed via modulesync
# https://github.com/voxpupuli/modulesync
# https://github.com/voxpupuli/modulesync_config
```

File: moduleroot/spec/acceptance/nodesets/docker/centos-5.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    hypervisor : docker
+    image: tianon/centos:5.10
+    docker_preserve_image: true
+    docker_cmd: '["/sbin/init"]'
+    docker_image_commands:
+      - 'yum install -y crontabs tar wget which'
+      - 'sed -i -e "/mingetty/d" /etc/inittab'
+CONFIG:
+  type: aio
+  log_level: debug
```
Comment (dhoppe) [inline moduleroot/spec/acceptance/nodesets/docker/centos-5.yml]: Please append:
```
...
# vim: syntax=yaml
```

**Comment by dhoppe** [comment]
Because we use that for other files as well.
```
dhoppe@macbook ..cts/puppet/voxpupuli/modulesync_config (git)-[master] % find . -type f | xargs grep vim
./config_defaults.yml:# vim: syntax=yaml
./Gemfile:# vim: syntax=ruby
./managed_modules.yml:# vim: syntax=yaml
./moduleroot/Gemfile:# vim: syntax=ruby
./moduleroot/Rakefile:# vim: syntax=ruby
./moduleroot/spec/acceptance/nodesets/centos-511-x64.yml:# vim: syntax=yaml
./moduleroot/spec/acceptance/nodesets/centos-66-x64-pe.yml:# vim: syntax=yaml
./moduleroot/spec/acceptance/nodesets/centos-66-x64.yml:# vim: syntax=yaml
./moduleroot/spec/acceptance/nodesets/centos-72-x64.yml:# vim: syntax=yaml
./moduleroot/spec/acceptance/nodesets/debian-78-x64.yml:# vim: syntax=yaml
./moduleroot/spec/acceptance/nodesets/debian-82-x64.yml:# vim: syntax=yaml
./moduleroot/spec/acceptance/nodesets/ubuntu-server-1204-x64.yml:# vim: syntax=yaml
./moduleroot/spec/acceptance/nodesets/ubuntu-server-1404-x64.yml:# vim: syntax=yaml
./moduleroot/spec/acceptance/nodesets/ubuntu-server-1604-x64.yml:# vim: syntax=yaml
./moduleroot/spec/classes/coverage_spec.rb:# vim: syntax=ruby
./moduleroot/spec/spec_helper.rb:# vim: syntax=ruby
./modulesync.yml:# vim: syntax=yaml
./Rakefile:# vim: syntax=ruby
```

**Comment by Yuav** [comment]
I've updated as per requirements now, but let the record show I think this should be fixed in vim, not every single file that could possibly be edited by vim.

**Pull request merged**